kafka_utils/util/protocol.py

```diff
@@ -12,46 +12,58 @@
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
 # limitations under the License.
-import struct
-
+import kafka.protocol.commit
 from kafka.protocol import KafkaProtocol
+from kafka.structs import ConsumerMetadataResponse
 from kafka.util import group_by_topic_and_partition
-from kafka.util import write_short_string
+from kafka.vendor import six
 
 
 class KafkaToolProtocol(KafkaProtocol):
 
     @classmethod
-    def encode_offset_commit_request_kafka(cls, client_id, correlation_id,
-                                           group, payloads):
+    def encode_offset_commit_request_kafka(cls, group, payloads):
         """
-        Encode some OffsetCommitRequest structs
+        Encode an OffsetCommitRequest struct
         Arguments:
-            client_id: string
-            correlation_id: int
             group: string, the consumer group you are committing offsets for
-            payloads: list of OffsetCommitRequest
+            payloads: list of OffsetCommitRequestPayload
         """
-        grouped_payloads = group_by_topic_and_partition(payloads)
-
-        message = []
-        message.append(cls._encode_message_header(
-            client_id, correlation_id,
-            KafkaProtocol.OFFSET_COMMIT_KEY,
-            version=2))
-        message.append(write_short_string(group))
-        message.append(struct.pack('>i', -1))   # ConsumerGroupGenerationId
-        message.append(write_short_string(''))  # ConsumerId
-        message.append(struct.pack('>q', -1))   # Retention time
-        message.append(struct.pack('>i', len(grouped_payloads)))
-
-        for topic, topic_payloads in grouped_payloads.items():
-            message.append(write_short_string(topic))
-            message.append(struct.pack('>i', len(topic_payloads)))
+        return kafka.protocol.commit.OffsetCommitRequest[2](
```

Comment:
Would it make sense to add https://github.com/dpkp/kafka-python/blob/master/kafka/protocol/commit.py#L64 as a comment to explain the 2, otherwise it requires a bit of context for a new person looking at this.